tests/ui/functional-struct-update/functional-struct-update-respects-privacy.rs

```diff
@@ -12,6 +12,7 @@ mod foo {
 
     pub fn make_secrets(a: u8, b: String) -> S {
         let val = unsafe { let p = COUNT.get(); let val = *p; *p = val + 1; val };
+        //~^ WARN creating a shared reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.